From 611c5c705db83d5b6577fba8b76bbe0f7778d291 Mon Sep 17 00:00:00 2001
From: Damien Thenot <damien.thenot@vates.tech>
Date: Fri, 28 Mar 2025 10:21:37 +0100
Subject: [PATCH] fix(FileSR): info were not correctly used

It would only use the last cowinfo list obtained instead of all of them
when listing, resulting in only the VDI of the last type being seen in
scan.

Signed-off-by: Damien Thenot <damien.thenot@vates.tech>
---
 drivers/FileSR.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/FileSR.py b/drivers/FileSR.py
index baf3c180..a75b6b7a 100755
--- a/drivers/FileSR.py
+++ b/drivers/FileSR.py
@@ -306,8 +306,8 @@ class FileSR(SR.SR):
 
             self.image_info.update(info)
 
-        for uuid in info.keys():
-            if info[uuid].error:
+        for uuid in self.image_info.keys():
+            if self.image_info[uuid].error:
                 raise xs_errors.XenError('SRScan', opterr='uuid=%s' % uuid)
 
             file_vdi = self.vdi(uuid)
@@ -315,7 +315,7 @@ class FileSR(SR.SR):
             self.vdis[uuid] = file_vdi
 
             # Get the key hash of any encrypted VDIs:
-            vdi_path = os.path.join(self.path, info[uuid].path)
+            vdi_path = os.path.join(self.path, self.image_info[uuid].path)
             key_hash = cowutil.getKeyHash(vdi_path)
             self.vdis[uuid].sm_config_override['key_hash'] = key_hash
 
